src/Views/AppListView.vala

```diff
@@ -40,20 +40,30 @@ namespace AppCenter.Views {
         }
 
         public override void add_packages (Gee.Collection<AppCenterCore.Package> packages) {
-            list_store.splice (0, 0, (GLib.Object[]) packages.to_array ());
-            list_store.sort ((GLib.CompareDataFunc<AppCenterCore.Package>) compare_packages);
+            foreach (var package in packages) {
+                add_row_for_package (package);
+            }
+
             if (current_visible_index < 20) {
                 show_more_apps ();
             }
         }
 
         public override void add_package (AppCenterCore.Package package) {
-            list_store.insert_sorted (package, (GLib.CompareDataFunc<AppCenterCore.Package>) compare_packages);
+            add_row_for_package (package);
+
             if (current_visible_index < 20) {
                 show_more_apps ();
             }
         }
 
+        private void add_row_for_package (AppCenterCore.Package package) {
+            // Don't show plugins in search and category views
```

Comment:
This is probably fine for now, but I think @danrabbit mentioned that better logic might be:

If the plugin has the same author as the original package, don't show it (because it's probably considered part of the app). But if the author of the plugin is different than the author of the app, show it so that third-party plugins remain searchable.